### PR TITLE
Enable 400G to 100G/40G speed change via minigraph for all platforms

### PR DESCRIPTION
**Pull request by anamehra**
Signed-off-by: anamehra anamehra@cisco.com

**- Why I did it**
There are chassis-packet and Single asic platforms which support this 400G to 100G/40G speed change via config.
Enabling this feature for all platforms which can support this. Keeping it enabled for all does not affect the platforms
which do not support this feature yet.

##### Work item tracking
- Microsoft ADO **(number only)**:

**- How I did it**
Removed switch_type and role type check.

**- How to verify it**
Loaded router with default 400G config. Loaded minigraph to convert 400G to 100G speed.

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106
- [ ] 202111
- [x] 202205
- [x] 202211

#### Tested branch (Please provide the tested image version)

- [ ] 
- [ ]

### CODE REVIEW

**Comment by anamehra** [comment]
Hi @abdosi , @arlakshm , raised this new PR due to CLA issue on other.

File: src/sonic-config-engine/minigraph.py [hunk 1]
```diff
@@ -1679,18 +1679,15 @@ def parse_xml(filename, platform=None, port_config_file=None, asic_name=None, hw
...
-                port_lanes =  ports[port_name].get('lanes', '').split(',')
-                # check if the 400g port has only 8 lanes
-                if len(port_lanes) != 8:
-                    continue
-                updated_lanes = ",".join(port_lanes[:4])
-                ports[port_name]['lanes'] = updated_lanes
-
+        # when the port speed is changes from 400g to 100g 
+        # update the port lanes, use the first 4 lanes of the 400G port to support 100G port
+        if port_default_speed == '400000' and port_png_speed == '100000':
```
Comment (abdosi) [inline src/sonic-config-engine/minigraph.py:1684]: add support of 40G also.

**Comment by abdosi** [comment]
@prgeor for viz.

File: src/sonic-config-engine/minigraph.py (hunk 1 above)
Comment (anamehra) [reply src/sonic-config-engine/minigraph.py]: Done!

**Comment by abdosi** [comment]
@anamehra can you check build failure ?

**Review by abdosi — APPROVED** [review]

**Comment by anamehra** [comment]
> @anamehra can you check build failure ?

Hi @abdosi , I see following error:
E: Failed to fetch http://packages.trafficmanager.net/snapshot/debian/20230423T000202Z/pool/main/g/gdb/gdb_10.1-1.7_amd64.deb  502  Bad Gateway [IP: 13.107.237.71 80]

Looks like some intermittent n/w issue?

**Comment by rlhui** [comment]
@snider-nokia please help review

**Review by prgeor — APPROVED** [review]

**Comment by abdosi** [comment]
@lguohan can you please help with merge of this

**Comment by anamehra** [comment]
Hi @abdosi , please help to get this merged in 202205. Thanks

**Comment by abdosi** [comment]
> @anamehra PR conflicts with 202305 branch

@anamehra : please create PR for 202305 branch:

**Comment by anamehra** [comment]
> > @anamehra PR conflicts with 202305 branch
> 
> @anamehra : please create PR for 202305 branch:

This is already part of 202305.

**Pull request merged**